### PR TITLE
Fad 3611

### PR DESCRIPTION
**Pull request by rajumsys**

### CODE REVIEW

File: test/dkim.js [hunk 1]
```diff
@@ -479,8 +478,31 @@ exports["Sign+verify tests"] = {
...
+    var dkimField = signMsg(mail, "node.ee", "dkim");
+    dkim.keyFromDNS = stubDNS;
+    dkim.DKIMVerify(dkimField + "\r\n" + mail, function(err, result) {
+      test.equal(err, null);
+      test.equal(result.sigs[0].warnings.length, 2);
+      test.ok(result.sigs[0].warnings[0].indexOf('is strongly recommended!'));
+      test.done();
+    });
+  },
+  "No warning if no header missing": function(test) {
```
Comment (rajumsys) [inline test/dkim.js:493]: i was talking about this test. it's failing now as it'll have warnings for missing `date` in `h`.

File: lib/dkim.js [hunk 2]
```diff
@@ -498,6 +501,24 @@ function DKIMVerify(email, callback, sigs) {
...
+    i;
+
+  //convert all headers to lowercase
+  for(i = 0; i < sig.h.length; i++){
+    lowerCaseHeaders.push(sig.h[i].toLowerCase());
+  }
+
+  for (i = 0; i < headersToWarnOnMissing.length; i++) {
+    if (lowerCaseHeaders.indexOf(headersToWarnOnMissing[i]) === -1) {
+      sig.warnings.push('Signing message ' + headersToWarnOnMissing[i] + ' is strongly recommended!');
```
Comment (yargevad) [inline lib/dkim.js]: How about `'Signing the ' + headersToWarnOnMissing[i] + ' header is strongly recommended!'`?

Comment (rajumsys) [reply lib/dkim.js]: no objection as i was also thinking `message` might be redundant or unnecessary.

**Review by yargevad — APPROVED** [review]
:shipit:

**Comment by rajumsys** [comment]
updated test. merging.

**Pull request merged**